pkg/text/text.go

```diff
@@ -0,0 +1,16 @@
+package text
+
+// TODO Text type:
+// TODO func NewText(text string, options ...TextOption) *Text {}
+// TODO func (t *Text) Text() string {}
+// TODO func (t *Text) Tokens() *TokenList {}
+// TODO func (t *Text) Words() *TokenList { return t.Tokens() }
+// TODO func (t *Text) Stems() *TokenList {}
+// TODO func (t *Text) Types() *TokenList { return t.Stems() }
+// TODO func (t *Text) Len() int {}
+// TODO func (t *Text) Runes() []rune {}
+// TODO func (t *Text) TokenCount() map[Token]int {}
+// TODO func (t *Text) TypeCount() map[Token]int {}
+// TODO func (t *Text) Vectorize() vector.Vector {}
+// TODO func (t *Text) Similarity(other *Text) float64 {}
```

Comment:
There should be a way to specify different similarity metrics here. I would probably do this with separate functions (e.g. `CosineSimilarity()` and `JaccardSimilarity()`).

Reply:
Good point, and we will probably want to do the same for vectorization so that we can support a few different vectorization methods (e.g. `CountVectorize()`, `TFIDFVectorize()`, `Word2Vectorize`)